appcenter-data/android/src/main/java/com/microsoft/appcenter/reactnative/data/AppCenterReactNativeDataModule.java

```diff
@@ -63,27 +63,58 @@ public String getName() {
     @ReactMethod
     public void read(String documentId, String partition, ReadableMap readOptionsMap, final Promise promise) {
         ReadOptions readOptions;
-        if (readOptionsMap.hasKey(TIME_TO_LIVE_KEY)) {
+        if(readOptionsMap == null) {
```

Comment:
@guperrot @dhei  I added a `null` check here to handle the `null` case.

Reply:
Missing a space.
```suggestion
        if (readOptionsMap == null) {
```

Reply:
Line 67 is a redundant because `readOptions` is initialized as `null` in Line 65.

```java
ReadOptions readOptions = null;
if (readOptionsMap != null && readOptionsMap.hasKey(TIME_TO_LIVE_KEY)) {
    // do something
} else {
    // do something else
}
```

Reply:
Yes there are only 2 cases, either use provided readOptions or pass null, with the code block from https://github.com/microsoft/appcenter-sdk-react-native/pull/620/files#r299605026.
Please simplify.

Reply:
Changed, I added [`getReadOptions`](https://github.com/microsoft/appcenter-sdk-react-native/pull/620/files#diff-52125d7be1f0c3bf1e2530f2d4a4d1d8R145) and [`getWriteOptions`](https://github.com/microsoft/appcenter-sdk-react-native/pull/620/files#diff-52125d7be1f0c3bf1e2530f2d4a4d1d8R155).